scripts/lib/CIME/XML/env_batch.py

```diff
@@ -439,53 +439,118 @@ def submit_jobs(self, case, no_batch=False, job=None, user_prereq=None,
         depid = OrderedDict()
         jobcmds = []
 
-        for job, dependency in jobs:
-            if dependency is not None:
-                deps = dependency.split()
-            else:
-                deps = []
-            dep_jobs = []
-            if user_prereq is not None:
-                dep_jobs.append(user_prereq)
-            for dep in deps:
-                if dep in depid.keys() and depid[dep] is not None:
-                    dep_jobs.append(str(depid[dep]))
-
-            logger.info("job {} depends on {}".format(job, dep_jobs))
-            result = self._submit_single_job(case, job,
-                                             dep_jobs=dep_jobs,
-                                             no_batch=no_batch,
-                                             skip_pnl=skip_pnl,
-                                             mail_user=mail_user,
-                                             mail_type=mail_type,
-                                             batch_args=batch_args,
-                                             dry_run=dry_run)
-            batch_job_id = str(alljobs.index(job)) if dry_run else result
-            depid[job] = batch_job_id
-            jobcmds.append( (job, result) )
-            if self._batchtype == "cobalt":
-                break
+        if resubmit_immediate:
+            num_submit = case.get_value("RESUBMIT") + 1
+            case.set_value("RESUBMIT", 0)
+            if num_submit <= 0:
+                num_submit = 1
+        else:
+            num_submit = 1
+
+        prev_job = None
+
+        for _ in range(num_submit):
+            for job, dependency in jobs:
+                if dependency is not None:
+                    deps = dependency.split()
+                else:
+                    deps = []
+                dep_jobs = []
+                if user_prereq is not None:
+                    dep_jobs.append(user_prereq)
+                for dep in deps:
+                    if dep in depid.keys() and depid[dep] is not None:
+                        dep_jobs.append(str(depid[dep]))
+                if prev_job is not None:
```

Comment:
I think that this is wrong - prev_job is only an explicit dependency if this is the first job in the current submit, otherwise dependancies are defined in env_batch.xml

Reply:
Isn't it the opposite? ie, prev_job is an explicit dependency of jobs which were submitted after the first one, which only explicitly depends on what's in env_batch.xml?
I've also realized I'm not handling the case where short term archiving jobs are submitted after the jobs complete, and need to think about how to deal with that and similar cases

Reply:
Dependancies between jobs in the variable "jobs" are defined in env_batch.xml
the only additional dependancy we need is between the last job in submit 1 and the first one in submit 2.